app/controllers/single_page_subscriptions_controller.rb

```diff
@@ -0,0 +1,63 @@
+class SinglePageSubscriptionsController < ApplicationController
+  include AccountHelper
+  include GovukPersonalisation::ControllerConcern
+
+  before_action do
+    head :not_found unless single_page_notifications_feature_enabled?
+  end
+
+  def show
+    content_item = GdsApi.content_store.content_item(topic).to_h
+    if logged_in?
+      user_id = GdsApi.account_api.get_user(
+        govuk_account_session: @account_session_header,
+      ).to_h["id"]
+
+      begin
+        GdsApi.email_alert_api.find_subscriber_by_govuk_account(
+          govuk_account_id: user_id,
+        )
+      rescue GdsApi::HTTPNotFound
+        GdsApi.email_alert_api.link_subscriber_to_govuk_account(
+          govuk_account_session: @account_session_header,
+        )
+      ensure
+        subscriber_list_slug = GdsApi.email_alert_api.find_or_create_subscriber_list(
+          { url: topic, name: content_item["title"] },
+        ).to_h.dig("subscriber_list", "slug")
+
+        GdsApi.account_api.put_email_subscription(
+          name: content_item["title"],
+          topic_slug: subscriber_list_slug,
+          govuk_account_session: @account_session_header,
+        )
```

Comment:
Just call email-alert-api directly to create the subscription, don't go via the account-api here.

The email subscription functionality in account-api is so that we can have a subscription with a consistent name like "brexit checker" which may actually refer to different topics at different times.  But that's not the case here.

Reply:
Would that be with [subscribe](https://github.com/alphagov/gds-api-adapters/blob/master/lib/gds_api/email_alert_api.rb#L76-L84) then?

So:
```
        GdsApi.account_api.subscribe(
          subscriber_list_id: subscriber_list_id,
          address:  "?" # guess we take this out of get_user call at the top?
        )
```

What values are we expecting for `frequency` and `skip_confirmation_email`?

Reply:
Use `GdsApi.email_alert_api`.  The call to `link_subscriber_to_govuk_account` above will return the subscriber, so you can get their address from that.

```ruby
subscriber_list = GdsApi.email_alert_api.find_or_create_subscriber_list(
  {
    url: topic,
    name: content_item["title"],
    content_id: content_item["content_id"],
  },
)

GdsApi.email_alert_api.subscribe(
      subscriber_list_id: subscriber_list.dig("subscriber_list", "id"),
      address: "take the address from link_subscriber_to_govuk_account",
      frequency: "daily",
      skip_confirmation_email: true,
    )
```

We've  been defaulting to daily subscriptions in other places, and the last email team wanted to phase out immediate subscriptions, so go with daily.